packages/amplify-frontend-ios/lib/frontend-config-creator.js

```diff
@@ -261,32 +261,45 @@ function getAppSyncConfig(appsyncResources, projectRegion) {
   // There can only be one appsync resource
   const appsyncResource = appsyncResources[0];
   const testMode = appsyncResource.testMode || false;
+  const { authConfig, securityType } = appsyncResource.output;
+  let authMode = '';
+
+  if (securityType) {
+    authMode = securityType;
+  } else if (authConfig) {
+    authMode = authConfig.defaultAuthentication.authenticationType;
+  }
+
+  const apiKey = authMode === 'API_KEY'
+    ? appsyncResource.output.GraphQLAPIKeyOutput
+    : undefined;
+
   const result = {
     AppSync: {
       Default: {
         ApiUrl: appsyncResource.output.GraphQLAPIEndpointOutput,
         Region: appsyncResource.output.region || projectRegion,
-        AuthMode: appsyncResource.output.securityType,
-        ApiKey:
-          appsyncResource.output.securityType === 'API_KEY'
-            ? appsyncResource.output.GraphQLAPIKeyOutput
-            : undefined,
-        ClientDatabasePrefix: `${appsyncResource.resourceName}_${appsyncResource.output.securityType}`,
+        AuthMode: authMode,
+        ApiKey: apiKey,
+        ClientDatabasePrefix: `${appsyncResource.resourceName}_${authMode}`,
       },
     },
   };
+
   if (testMode) {
     result.AppSync.Default.DangerouslyConnectToHTTPEndpointForTesting = true;
   }
 
-  const additionalAuths = appsyncResource.output.additionalAuthenticationProviders || [];
-  additionalAuths.forEach((authType) => {
-    const apiName = `${appsyncResource.resourceName}_${authType}`;
+  const additionalAuths = appsyncResource.output.authConfig.additionalAuthenticationProviders || [];
```

Comment:
so I'm trying to initialize environment with new cli version and it fails on line 293, can anyone help? why is this not backward compatible?

$ amplify env add
Note: It is recommended to run this command from the root of your app directory
? Do you want to use an existing environment? Yes
? Choose the environment you would like to use: production
Using default provider  awscloudformation
✔ Initialized provider successfully.
✖ There was an error initializing your environment.
init failed
TypeError: Cannot read property 'additionalAuthenticationProviders' of undefined
    at getAppSyncConfig (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-frontend-ios/lib/frontend-config-creator.js:293:61)

Reply:
@attilah